### PR TITLE
chore: fonction aws api

### PR DESCRIPTION
**Pull request by vincentc-afk**

### CODE REVIEW

File: Source/Aws/aws-api.env [hunk 1]
```diff
@@ -77,25 +77,113 @@ _t4dSrcAwsInstanceQuerry(){
...
+# v1.8.0
+###### _t4dSrcAwsImageDescribe
+#   - Name:
+#       _t4dSrcAwsImageDescribe
+###
+#   - Synopsis:
+#        Describe image id
+###
+#   - Usage:
+#       * {Name} <option>  [arguments]        = {One-Line Description}
```
Comment (QuentinNIO) [inline Source/Aws/aws-api.env:144]: doc

File: Source/Aws/aws-api.env [hunk 2]
```diff
@@ -77,25 +77,113 @@ _t4dSrcAwsInstanceQuerry(){
...
+# v1.8.0
+###### _t4dSrcAwsSubNetDescribe
+#   - Name:
+#       _t4dSrcAwsSubNetDescribe
+###
+#   - Synopsis:
+#        Describe subnet id
+###
+#   - Usage:
+#       * {Name} <option>  [arguments]        = {One-Line Description}
```
Comment (QuentinNIO) [inline Source/Aws/aws-api.env:171]: doc

File: Source/Manifest/_main_.env [hunk 3]
```diff
@@ -0,0 +1 @@
+source $T4D_SOURCE_PATH/Manifest/manifest-xml.env
```
Comment (QuentinNIO) [inline Source/Manifest/_main_.env:1]: add new line at end of file

File: Source/Common/xml.env [hunk 4]
```diff
@@ -76,13 +70,11 @@ _t4dFindMarkerInXML(){
...
-        cat ${_filename}.xml | grep -B2000 "</manifest>" | grep "<${_type} " | grep -Eo " ${_marker}=.{0,200}" | cut -d '"' -f2
-    else
-        _t4dDebugLog $pwarning "_t4dFindMarkerInXML - ${_filename}.xml File Not Found"
-        return 1
-    fi
+    local _filename="${3}"
+
+    _t4dManifestXmlGetMarker "$_type" "$_marker" "$_filename"
+    _t4dDebugLog $pwarning "This is deprecated, please use _t4dManifestXmlGetMarker"
 }
```
Comment (QuentinNIO) [inline Source/Common/xml.env:80]: add new line at end of file

File: Source/Manifest/manifest-xml.env [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+########### CODE BLOC #########
+    local _t4dErrorCode=1
+
+    local _type=$1
+    local _nameType=$2
+    local _name=$3
+    local _filename="$4"
+
+    if [[ -e "${_filename}" ]]; then
+        _filename="$(cat $_filename)"
```
Comment (QuentinNIO) [inline Source/Manifest/manifest-xml.env:30]: add elif -e "${_filename}.xml"

then _Content="$(cat $_filename.xml)"

File: Source/Manifest/manifest-xml.env [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+    fi
+
+    echo "$_filename" | grep "<${_type} " | grep " $_nameType=\"${_name}\""
```
Comment (QuentinNIO) [inline Source/Manifest/manifest-xml.env:33]: replace _fileName with _Content as it does not describe var properly

File: Source/Manifest/manifest-xml.env [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+########### CODE BLOC #########
+    local _t4dErrorCode=1
+
+    local _type=$1
+    local _nameType=$2
+    local _name=$3
+    local _attribute=$4
+    local _filename="$5"
+
+    if [[ -e "${_filename}" ]]; then
```
Comment (QuentinNIO) [inline Source/Manifest/manifest-xml.env:67]: you copy paste those content, instead, please consider creating a specific function that will return xml content and check that sended content/file is a manifest

File: Source/Manifest/manifest-xml.env [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+###### DOCUMENTATION BLOC #####
+########### CODE BLOC #########
+    local _t4dErrorCode=1
+
+    local _type=$1
+    local _marker=$2
+    local _filename="$3"
+
+
+    if [[ -e "${_filename}" ]]; then
```
Comment (QuentinNIO) [inline Source/Manifest/manifest-xml.env:104]: same as bellow

File: Source/Manifest/manifest-xml.env [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+        _filename="$(cat $_filename)"
+    fi
+
+    echo "$_filename" | grep "<${_type} " | grep -Eo " ${_marker}=.*" | cut -d '"' -f2
+
+###############################
+}
+
+_t4dManifestXmlStart(){
```
Comment (QuentinNIO) [inline Source/Manifest/manifest-xml.env:113]: start is also edit ?

File: Source/Manifest/manifest-xml.env [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+            ;;
+            -type|-t)
+                _Type="$2"
+                _Res="<${_Type}${_Res}"
+                shift
+            ;;
+            *)
+                _Arg="$1"
+                local _Marker=$(echo $_Arg | cut -d"=" -f1)
+                local _Data=$(echo $_Arg | cut -d"=" -f2)
```
Comment (QuentinNIO) [inline Source/Manifest/manifest-xml.env:238]: what if data is containing `=` ? :)

File: Source/Manifest/manifest-xml.env [hunk 11]
```diff
@@ -0,0 +1,261 @@
...
+                _Res="${_Res} $_Marker=\"$_Arg\""
+            ;;
+        esac
+        shift
+    done
+
```
Comment (QuentinNIO) [inline Source/Manifest/manifest-xml.env:244]: what if user is adding -type at the end of their command ? this should be way stronger than it is right now.

Comment (QuentinNIO) [reply Source/Manifest/manifest-xml.env:244]: `_t4dManifestXmlSetLine "type=kvm" "time=$(date +%s)" -type provider "another=oui" -type override`

type this, and it to example and fix bugs :)

**Review by QuentinNIO — CHANGES_REQUESTED** [review]

File: Source/Manifest/manifest-xml.env (hunk 9 above)
Comment (vincentc-afk) [reply Source/Manifest/manifest-xml.env:113]: start for edit

File: Source/Manifest/manifest-xml.env (hunk 10 above)
Comment (vincentc-afk) [reply Source/Manifest/manifest-xml.env:238]: in Xml, it is impossible to use the equal sign, or even special characters, the only way is to use their code, so not possible to find an equal in their data

https://dvteclipse.com/documentation/svlinter/How_to_use_special_characters_in_XML.3F.html

File: Source/Manifest/manifest-xml.env (hunk 9 above)
Comment (QuentinNIO) [reply Source/Manifest/manifest-xml.env]: did you update this to open/close ?

**Review by QuentinNIO — CHANGES_REQUESTED** [review]

**Review by QuentinNIO — CHANGES_REQUESTED** [review]

File: Source/Manifest/manifest-xml.env [hunk 12]
```diff
@@ -0,0 +1,296 @@
...
+#   - Note:
+#       <Required>
+#       [Optionnal]
+#       {XXXXXXX}       = Value To Be Changed, if {URL} then replace {URL} with a valid URL
+###
+###### DOCUMENTATION BLOC #####
+########### CODE BLOC #########
+    local _t4dErrorCode=1
+
+    local _File="$1"
```
Comment (QuentinNIO) [inline Source/Manifest/manifest-xml.env:287]: _Content

**Review by QuentinNIO — CHANGES_REQUESTED** [review]

**Review by QuentinNIO — APPROVED** [review]

**Pull request merged**